### PR TITLE
Contrib dca easyconfig

### PR DESCRIPTION
**Pull request by gbalduzz**
I hope this time I issued the pull request against the correct branch and followed the correct procedure.
Cheers, G.B.

### CODE REVIEW

**Comment by gppezzi** [comment]
I've issued a PR to @gbalduzz's branch to fix the file name of the DCA easyconfig files.

As for the rest, it looks good.

File: easybuild/easyconfigs/g/GSL/GSL-2.1-CrayGNU-2015.11.eb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Built with EasyBuild version 2.8.1 on 2016-06-16_13-37-02
```
Comment (gppezzi) [inline easybuild/easyconfigs/g/GSL/GSL-2.1-CrayGNU-2015.11.eb]: Can you please remove this comment?

File: easybuild/easyconfigs/g/GSL/GSL-2.1-CrayGNU-2015.11.eb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+name = 'GSL'
+version = '2.1'
+
+homepage = 'http://www.gnu.org/software/gsl/'
+description = """The GNU Scientific Library (GSL) is a numerical library for C and C++ programmers.
+ The library provides a wide range of mathematical routines such as random number generators, special functions
+ and least-squares fitting."""
+
+toolchain = {'name': 'CrayGNU', 'version': '2015.11'}
+toolchainopts = {'opt': True, 'optarch': True, 'unroll': True, 'pic': True}
```
Comment (gppezzi) [inline easybuild/easyconfigs/g/GSL/GSL-2.1-CrayGNU-2015.11.eb]: Can you check on the build logs if this toolchain option 'unroll' is translated to the correct flags (CFLAGS variable for example)?

I'm not sure if this is implemented on CrayGNU.. If it is not working you will probably see `-True` added to the flags and in that case I think it is better to remove `'unroll': True` from the easyconfig file.

**Comment by gppezzi** [comment]
Test report by @gppezzi
**SUCCESS**
Build succeeded for 7 out of 7 (6 easyconfigs in this PR)
daint101 - Linux SLES 11_SP3, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.6
See https://gist.github.com/6e11526bd638805b7a5fb77363627355 for a full test report.

File: easybuild/easyconfigs/g/GSL/GSL-2.1-CrayGNU-2015.11.eb (hunk 2 above)
Comment (gppezzi) [inline easybuild/easyconfigs/g/GSL/GSL-2.1-CrayGNU-2015.11.eb]: I've just tested and the 'unroll' option seems to be working.

Comment (gbalduzz) [inline easybuild/easyconfigs/g/GSL/GSL-2.1-CrayGNU-2015.11.eb]: Sorry, I was too slow. Thanks for checking!

File: easybuild/easyconfigs/m/magma/magma-2.0.0-CrayGNU-2015.11.eb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+skipsteps = ['configure']
+
+builddependencies = [ ('cudatoolkit/7.0.28-1.0502.10742.5.1', EXTERNAL_MODULE) ]
+ 
+sanity_check_paths={
+    'files': ['lib/libmagma.so', 'lib/libmagma.a'],
+    'dirs': ['include'],
+}
+
+patches = [('magma-2.0.0.patch')]
```
Comment (boegel) [inline easybuild/easyconfigs/m/magma/magma-2.0.0-CrayGNU-2015.11.eb]: please move this up below `sources`/`source_urls`

File: easybuild/easyconfigs/g/GSL/GSL-2.1-CrayGNU-2015.11.eb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+source_urls = [GNU_SOURCE]
+sources = [SOURCELOWER_TAR_GZ]
+
+configopts = "--with-pic"
+
```
Comment (boegel) [inline easybuild/easyconfigs/g/GSL/GSL-2.1-CrayGNU-2015.11.eb]: no `sanity_check_paths`?

File: easybuild/easyconfigs/d/DCA++/DCA++-1.0-CrayGNU-2015.11-cuda.eb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+easyblock = 'Bundle'
+
+name = 'DCA++'
+version = '1.0'
+versionsuffix = '-cuda'
+
+homepage = 'http://www.itp.phys.ethz.ch/research/comp/computation.html'
+description = """The DCA++ software project is a C++ implementation of the dynamical cluster approximation (DCA) and its DCA+ extension. It aims to solve lattice models of strongly correlated electron systems. This module bundles all the dependencies for the CPU+GPU version."""
```
Comment (boegel) [inline easybuild/easyconfigs/d/DCA++/DCA++-1.0-CrayGNU-2015.11-cuda.eb]: too long, please wrap

File: easybuild/easyconfigs/d/DCA++/DCA++-1.0-CrayGNU-2015.11-cpu.eb [hunk 6]
```diff
@@ -0,0 +1,21 @@
+easyblock = 'Bundle'
+
+name = 'DCA++'
+version = '1.0'
+versionsuffix = '-cpu'
+
+homepage = 'http://www.itp.phys.ethz.ch/research/comp/computation.html'
+description = """The DCA++ software project is a C++ implementation of the dynamical cluster approximation (DCA) and its DCA+ extension. It aims to solve lattice models of strongly correlated electron systems. This module bundles all the dependencies for the CPU-only version."""
```
Comment (boegel) [inline easybuild/easyconfigs/d/DCA++/DCA++-1.0-CrayGNU-2015.11-cpu.eb]: too long, please wrap

File: easybuild/easyconfigs/m/magma/magma-2.0.0-CrayGNU-2015.11.eb [hunk 7]
```diff
@@ -0,0 +1,28 @@
+easyblock = "ConfigureMake"
+
+name = 'magma'
+version = '2.0.0'
+
+homepage = 'http://icl.cs.utk.edu/magma/'
+description = "The MAGMA project aims to develop a dense linear algebra library similar to LAPACK but for heterogeneous/hybrid architectures, starting with current Multicore+GPU systems."
```
Comment (boegel) [inline easybuild/easyconfigs/m/magma/magma-2.0.0-CrayGNU-2015.11.eb]: too long, please wrap

File: easybuild/easyconfigs/m/magma/magma-2.0.0-CrayGNU-2015.11.eb [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+toolchain = {'name': 'CrayGNU', 'version': '2015.11'}
+toolchainopts = {'pic': True}
+
+preinstallopts = "export EBINSTALLPREFIX=%(installdir)s && "
```
Comment (boegel) [inline easybuild/easyconfigs/m/magma/magma-2.0.0-CrayGNU-2015.11.eb]: please move this down below `skipsteps`

File: easybuild/easyconfigs/m/magma/magma-2.0.0-CrayGNU-2015.11.eb [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+sources = [SOURCE_TAR_GZ]
+source_urls = ['http://icl.cs.utk.edu/projectsfiles/magma/downloads/']
+
+skipsteps = ['configure']
+
+builddependencies = [ ('cudatoolkit/7.0.28-1.0502.10742.5.1', EXTERNAL_MODULE) ]
```
Comment (boegel) [inline easybuild/easyconfigs/m/magma/magma-2.0.0-CrayGNU-2015.11.eb]: please move this up above `skipsteps`

File: easybuild/easyconfigs/m/magma/magma-2.0.0.patch [hunk 10]
```diff
@@ -0,0 +1,106 @@
+diff -ru magma-2.0.0.orig//Makefile magma-2.0.0/Makefile
```
Comment (boegel) [inline easybuild/easyconfigs/m/magma/magma-2.0.0.patch]: please add a comment on top of this patch explaining what it does, and who the author is

**Comment by gppezzi** [comment]
Test report by @gppezzi
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
daint104 - Linux SLES 11_SP3, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.6
See https://gist.github.com/d17acd81f5d863a3e5dc0699991a3eb0 for a full test report.

**Comment by gppezzi** [comment]
@boegel I hope I didn't miss any of your remarks, now it should be good to go

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Going in, thanks @gbalduzz!

**Pull request merged**